New package: Atlisp.Atlisp version 1.0.0

Dragon1573 opened the pull request:
## Checklist for Pull Requests

- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Resolve #248227

## Manifests

- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.10 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.10.0)?

Note: `<path>` is the directory's name containing the manifest you're submitting.

## Manually verified

```yaml
InstallerType: zip
NestedInstallerType: portable
NestedInstallerFiles:
- RelativeFilePath: atlisp/atlisp.exe
Installers:
- Architecture: x64
  InstallerUrl: https://atlisp.cn/download/atlisp-v1.0.0.zip
  InstallerSha256: 49EB9790F03C636C852C8C5AAEFE220A9C0442DAEF249193514B939F882BE882
```

```text
--> Installing WinGet
--> Disabling safety warning when running installers
Tip: you can type 'Update-EnvironmentVariables' to update your environment variables, such as after installing a new software.

--> Configuring Winget
已启用管理员设置 'LocalManifestFiles'。
已启用管理员设置 'LocalArchiveMalwareScanOverride'。
已启用管理员设置 'ProxyCommandLineOptions'。
将管理员设置 'DefaultProxy' 设置为 'http://alist.dragon1573.local:57890'。

--> Installing the Manifest 1.0.0

已找到 @lisp CAD应用云 [Atlisp.Atlisp] 版本 1.0.0
此应用程序由其所有者授权给你。
Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
正在下载 https://atlisp.cn/download/atlisp-v1.0.0.zip
  ██████████████████████████████  10.2 MB / 10.2 MB
已成功验证安装程序哈希
正在提取存档...
已成功提取存档
正在启动程序包安装...
添加了命令行别名： "atlisp"
已修改路径环境变量；重启 shell 以使用新值。
已成功安装

--> Refreshing environment variables

--> Comparing ARP Entries

DisplayName  DisplayVersion Publisher     ProductCode                  Scope
-----------  -------------- ---------     -----------                  -----
@lisp CAD应用云 1.0.0          河北好逐易信息技术有限公司 Atlisp.Atlisp__DefaultSource User

PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> atlisp.exe -h
usage: atlisp.exe command <pkgname/keystring>
Version: 1.0.0
@lisp是一个运行于 AutoCAD、中望CAD、浩辰CAD及类似兼容的CAD系统中的应用管理器。

command       function:
  pull        安装@lisp包 到 CAD
  install     安装@lisp Core 到 CAD
  list        列已安装的@lisp包
  search      联网搜索 @lisp 包

options:
  -h, --help  show this help message and exit
  -v, --version 显示当前安装的版本号

PS C:\Users\WDAGUtilityAccount\Desktop\winget-pkgs> atlisp.exe -v
atlisp version 1.0.0
```

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/248242)

Dragon1573 committed:
New package: Atlisp.Atlisp version 1.0.0

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

vitalgg commented:
`安装最新的 PowerShell，了解新功能和改进！https://aka.ms/PSWindows

PS C:\Users\Administrator> winget  install --manifest ..\vitalgg\projects\winget-pkgs\manifests\a\atlisp\atlisp\1.0.0\
已找到 atlisp [atlisp.atlisp] 版本 1.0.0
此应用程序由其所有者授权给你。
Microsoft 对第三方程序包概不负责，也不向第三方程序包授予任何许可证。
正在下载 https://atlisp.cn/download/atlisp-1.0.0.zip
  ██████████████████████████████  10.2 MB / 10.2 MB
已成功验证安装程序哈希
正在提取存档...
已成功提取存档
正在启动程序包安装...
添加了命令行别名： "atlisp"
已修改路径环境变量；重启 shell 以使用新值。
已成功安装
PS C:\Users\Administrator>`



# Created with YamlCreate.ps1 v2.4.6 $debug=NVS1.CRLF.5-1-22621-4249.Win32NT
# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.10.0.schema.json

PackageIdentifier: atlisp.atlisp
PackageVersion: 1.0.0
InstallerLocale: zh-CN
InstallerType: zip
InstallerSuccessCodes:
- "1"
Commands:
- atlisp
Protocols:
- https
FileExtensions:
- url
ReleaseDate: 2025-04-15
Installers:
- Architecture: x64
  NestedInstallerType: portable
  NestedInstallerFiles:
  - RelativeFilePath: atlisp.exe
    PortableCommandAlias: atlisp
  InstallerUrl: https://atlisp.cn/download/atlisp-1.0.0.zip
  InstallerSha256: 1B80A7585317BE07D1D0A3A95C35B30910B10B65F95B6F6BAC7173BCA2CC3306
ManifestType: installer
ManifestVersion: 1.10.0

Dragon1573 commented:
> [!IMPORTANT]
> 
> 单反引号对仅适用于嵌入在行内部的代码片段，对于多行代码片段请使用三反引号对以确保其被正确渲染。
> 
> 确保在评论前查看 Preview 板块以保证相关评论内容格式正确。

Dragon1573 commented:
@stephengillie, further investigation required. 🙏🏼 

I've [already validated](https://github.com/microsoft/winget-pkgs/pull/248242#issue-2995150391) on my local _Windows Sandbox_ instance. Can't figure out what was wrong ...

![image](https://github.com/user-attachments/assets/0d62dd4d-4d94-46ab-b7a6-509e6dcef5cd)

vitalgg commented:
前次的安装错误是可执行路径与zip包中的路径不一致导致的。

Dragon1573 commented on manifests/a/Atlisp/Atlisp/1.0.0/Atlisp.Atlisp.installer.yaml:
```diff
@@ -0,0 +1,21 @@
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: atlisp/atlisp.exe
```
```suggestion
- RelativeFilePath: atlisp.exe
InstallerSuccessCodes:
- "1"
Commands:
- atlisp
Protocols:
- https
FileExtensions:
- url
```

Dragon1573 commented on manifests/a/Atlisp/Atlisp/1.0.0/Atlisp.Atlisp.installer.yaml:
```diff
@@ -0,0 +1,21 @@
+- silent
+- silentWithProgress
+UpgradeBehavior: uninstallPrevious
+ReleaseDate: 2025-04-11
```
```suggestion
ReleaseDate: 2025-04-15
```

Dragon1573 committed:
Apply suggestions from code review

- Correction of nested file path and related properties.

Dragon1573 committed:
Update manifest: Atlisp.Atlisp version 1.0.0

- Update the installer URL and SHA256 hash to match the new version.
- Correction of propery type for InstallerSuccessCodes from string to integer.

Signed-off-by: Dragon1573 <49941141+Dragon1573@users.noreply.github.com>

Dragon1573 commented:
Further investigation required. 🙏🏼 

![image](https://github.com/user-attachments/assets/eff0b19c-3875-4fac-a7eb-bed3e1b03655)

ItzLevvie commented:
@wingetbot run

stephengillie commented:
Automatic Validation ended with:
>  Installation failed with exit code -1978335187 2025-04-16 04:47:13.434 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerSharedLib\SHA256.cpp(144)\WindowsPackageManager.dll!00007FF80E26273F: (caller: 00007FF80E2AAEC4) Exception(1) tid(1ca0) 8A150040  2025-04-16 04:47:31.283 [FAIL] Installer failed security check. Url: https://atlisp.cn/download/atlisp-1.0.0.zip Result: 0x80004005    | Hex | Dec | Inverted Dec | Symbol | Description |   | --- | --- | --- | --- | --- |   | 80004005 | -2147467259 | 2147500037 | E_Fail | Resource missing |   | 8A15002D | -1978335187 | 2316632109 | APPINSTALLER_CLI_ERROR_INSTALLER_SECURITY_CHECK_FAILED | Installer failed security check.  Another script should retry this about every 18 hours. <!-- [Policy] Needs-Attention [Policy] Validation-Defender-Error --> |   | 8A150040 | -1978335168 | 2316632128 | APPINSTALLER_CLI_ERROR_STREAM_READ_FAILURE | Error reading from the stream | 
 
 | Hex | Dec | Inverted Dec | Symbol | Description | 
 | --- | --- | --- | --- | --- | 
 | 00000012 | 18 | -4294967278 | ERROR_NO_MORE_FILES | There are no more files. | 
 | 80004005 | -2147467259 | 2147500037 | E_Fail | Resource missing | 
 | 80004005 | -2147467259 | 2147500037 | E_Fail | Resource missing | 
 | 80004005 | -2147467259 | 2147500037 | E_Fail | Resource missing | 
 | 8A15002D | -1978335187 | 2316632109 | APPINSTALLER_CLI_ERROR_INSTALLER_SECURITY_CHECK_FAILED | Installer failed security check.  Another script should retry this about every 18 hours. <!-- [Policy] Needs-Attention [Policy] Validation-Defender-Error --> | 
 | 8A15002D | -1978335187 | 2316632109 | APPINSTALLER_CLI_ERROR_INSTALLER_SECURITY_CHECK_FAILED | Installer failed security check.  Another script should retry this about every 18 hours. <!-- [Policy] Needs-Attention [Policy] Validation-Defender-Error --> | 
 | 8A15002D | -1978335187 | 2316632109 | APPINSTALLER_CLI_ERROR_INSTALLER_SECURITY_CHECK_FAILED | Installer failed security check.  Another script should retry this about every 18 hours. <!-- [Policy] Needs-Attention [Policy] Validation-Defender-Error --> | 
 | 8A150040 | -1978335168 | 2316632128 | APPINSTALLER_CLI_ERROR_STREAM_READ_FAILURE | Error reading from the stream | 
 | 8A150040 | -1978335168 | 2316632128 | APPINSTALLER_CLI_ERROR_STREAM_READ_FAILURE | Error reading from the stream | 
 | 8A150040 | -1978335168 | 2316632128 | APPINSTALLER_CLI_ERROR_STREAM_READ_FAILURE | Error reading from the stream | 


(Automated response - build 1042.)

Dragon1573 commented:
> The package manager bot was blocked from installing one of the installers listed in the url field, and cannot continue.
> 
> The application included in this pull request failed to pass the [Installers Scan](https://docs.microsoft.com/en-us/windows/package-manager/package/winget-validation) test. This test is designed to ensure that the application installs on all environments without warnings.
> 
> Please check to ensure the installer URL is correct and update the URL and the Hash if a change is made.
> 
> If the package is mistakenly flagged as malware, you can [submit the installer(s) to the Microsoft Defender team for analysis](https://www.microsoft.com/en-us/wdsi/filesubmission).
> 
> For more details on this error, see [Defender errors](https://learn.microsoft.com/en-us/windows/package-manager/package/repository#error-labels).
> 
> Template: msftbot/validationError/installers/validationDefender

CC @vitalgg, Microsoft Defender 扫描后认为其有害。可以考虑将安装包提交给他们去消除假阳性（估计要等很久）。

你这个项目是用 Pyinstaller 构建的么？有不少使用了这个框架做封装的都有被误标的情况。或许可以尝试一下 Nuitka 之类的其他构建封装技术？

stephengillie commented:
@wingetbot run

vitalgg commented:
> CC @vitalgg, Microsoft Defender 扫描后认为其有害。可以考虑将安装包提交给他们去消除假阳性（估计要等很久）。
> 
> 你这个项目是用 Pyinstaller 构建的么？有不少使用了这个框架做封装的都有被误标的情况。或许可以尝试一下 Nuitka 之类的其他构建封装技术？

是用pyinstaller 构建的。好的，我试试nuitka .

vitalgg commented:
用nuitka 打包成单文件模式，本地defender也报有害，standalong模式下link文件又不能正常运行。
这个是用已发布的pip包编译的。不行的话我用其它语言重构一下吧。

Dragon1573 commented:
> 用nuitka 打包成单文件模式，本地defender也报有害，standalong模式下link文件又不能正常运行。 这个是用已发布的pip包编译的。不行的话我用其它语言重构一下吧。

还有一种方式，PyInstaller / Nuitka 构建时不使用单文件，构建后输出应该是一个包含了可执行文件及其必要动态链接库的目录，对这个目录再用 InnoSetup 等工具构造为安装向导程序。这种方式可能会比较麻烦。

本仓库的 Validation Pipeline 运行在 Microsoft Azure 上，抗病毒引擎统一使用 Windows Defender 。如果 WD 报毒了，我们（社区成员）确实无能为力。有不少应用因为这个假阳性误报都没法顺利入库收录😞

-----

@stephengillie,

the original author / publisher of this package tried re-packing the application with Nuitka instead of Pyinstaller, but Windows Defender still flagged it as a malware. We're waiting for the author "rewrite" the entire project with other programming language, to avoid false positive malware report.

vitalgg commented:
mingw 编译的程序与误报！！！
只有微软自家的开发语言才能过吗？

新生成的 atlisp.atlisp.installer.yaml 内容如下：
```yaml
PackageIdentifier: atlisp.atlisp
PackageVersion: 1.0.0
InstallerLocale: zh-CN
InstallerType: zip
InstallerSuccessCodes:
- "1"
Commands:
- atlisp
Protocols:
- https
FileExtensions:
- url
ReleaseDate: 2025-04-17
Installers:
- Architecture: x64
  NestedInstallerType: portable
  NestedInstallerFiles:
  - RelativeFilePath: atlisp.exe
    PortableCommandAlias: atlisp
  InstallerUrl: https://atlisp.cn/download/atlisp-1.0.0.zip
  InstallerSha256: B6234DBCE7B029FB570F72F408256B7DF025C4B2308574DC829B58C38816E3DF
ManifestType: installer
ManifestVersion: 1.10.0
```

Dragon1573 commented:
✅ 本地复核，可复现误报。

![image](https://github.com/user-attachments/assets/428b0cbb-8714-4946-b8de-acb0d295323d)

Dragon1573 commented:
> 为什么执行文件 atlisp.exe 没问题，只是压缩包 zip文件有问题啊。

当前 `winget.exe` 对 InnoSetup / Burn / WiX / MSI / MSIX 等格式的安装向导程序提供了完善的支持，是否可以考虑使用 InnoSetup 打包编译产物，不直接使用 ZIP 压缩发行？

DuckDuckStudio commented:
提交分析过了吗 👀

DuckDuckStudio commented:
我觉得我们可以让它在验证管道中多试几次（我的软件包就是试着试着就过了XD）

ItzLevvie commented:
@wingetbot run

DuckDuckStudio commented on manifests/a/Atlisp/Atlisp/1.0.0/Atlisp.Atlisp.installer.yaml:
```diff
@@ -0,0 +1,29 @@
+Installers:
+- Architecture: x64
+  InstallerUrl: https://atlisp.cn/download/atlisp-1.0.0.zip
+  InstallerSha256: 1B80A7585317BE07D1D0A3A95C35B30910B10B65F95B6F6BAC7173BCA2CC3306
```
```suggestion
  InstallerSha256: EED90F36A4CBAE3997150F1A4138385D5C4E1B92E2E982247813DFED2F765A5A
```

更新哈希值后再试一次？

Dragon1573 committed:
Apply suggestions from code review

- Correction of installer SHA value.

Co-authored-by: 鸭鸭「カモ」 <89643991+DuckDuckStudio@users.noreply.github.com>

Dragon1573 commented on manifests/a/Atlisp/Atlisp/1.0.0/Atlisp.Atlisp.installer.yaml:
```diff
@@ -0,0 +1,29 @@
+InstallerType: zip
+NestedInstallerType: portable
+NestedInstallerFiles:
+- RelativeFilePath: atlisp.exe
```
```suggestion
- RelativeFilePath: atlisp/atlisp.exe
```

Dragon1573 committed:
Apply suggestions from code review

- Correction of nested file path.

stephengillie commented:
Manual Validation ended with:
> 
> PS C:\Users\User\Desktop> C:\Users\User\AppData\Local\Microsoft\WinGet\Links\atlisp.exe --version
> CLR: Assert failure(PID 5692 [0x0000163c], Thread: 8644 [0x21c4]): !AreShadowStacksEnabled() || UseSpecialUserModeApc()
>     File: D:\a\_work\1\s\src\coreclr\vm\threads.cpp:7938 Image:
> C:\Users\User\AppData\Local\Microsoft\WinGet\Links\atlisp.exe
> PS C:\Users\User\Desktop> C:\Users\User\AppData\Local\Microsoft\WinGet\Links\atlisp.exe
> CLR: Assert failure(PID 10204 [0x000027dc], Thread: 6212 [0x1844]): !AreShadowStacksEnabled() || UseSpecialUserModeApc()
>     File: D:\a\_work\1\s\src\coreclr\vm\threads.cpp:7938 Image:
> C:\Users\User\AppData\Local\Microsoft\WinGet\Links\atlisp.exe
> PS C:\Users\User\Desktop>

(Automated response - build 1049.)

Dragon1573 commented:
It's so weird and unpredictable ... 😕
